pkg/frontend/querymiddleware/roundtrip.go

```diff
@@ -136,6 +137,19 @@ type MetricsQueryHandler interface {
 	Do(context.Context, MetricsQueryRequest) (Response, error)
 }
 
+// LabelsHandlerFunc is like http.HandlerFunc, but for LabelsQueryHandler.
+type LabelsHandlerFunc func(context.Context, LabelsQueryRequest) (Response, error)
+
+// Do implements LabelsQueryHandler.
+func (q LabelsHandlerFunc) Do(ctx context.Context, req LabelsQueryRequest) (Response, error) {
+	return q(ctx, req)
+}
+
+// LabelsQueryHandler is like http.Handle, but specifically for Prometheus label names and values calls.
```

Comment:
Typo?

```suggestion
// LabelsQueryHandler is like http.Handler, but specifically for Prometheus label names and values calls.
```

Reply:
It was copied from MetricsQueryHandler, but it does seem to be a typo so I updated both,